### PR TITLE
Add some test expressions

### PR DESCRIPTION
**Pull request by Efnilite**
### Description
Adds some test expressions which I felt were missing.

- `test world` can now be used instead of `world "world"`
- `test location` can now be used instead of `spawn of world "world"`
- `test block` can now be used instead of `block at spawn of world "world"`
- All changed tests have no changed logic, except the block iterator test. This was changed to not fail since some other test forgot to reset the block to air, so it would've been a chest, failing the test.

Also adds `event-world` event value.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Review by Fusezion — APPROVED** [review]

**Comment by NotSoDelayed** [comment]
There’s `event-location` in tests
https://github.com/SkriptLang/Skript/blob/ba32d7ad48edb8e33d5f5d2e44d03bba94ffb2de/src/main/java/ch/njol/skript/test/runner/EvtTestCase.java#L45-L59

**Comment by Efnilite** [comment]
> There’s `event-location` in tests
> 
> https://github.com/SkriptLang/Skript/blob/ba32d7ad48edb8e33d5f5d2e44d03bba94ffb2de/src/main/java/ch/njol/skript/test/runner/EvtTestCase.java#L45-L59

this is used in like 5 tests, also `test-location` looks nicer :)

**Comment by TheLimeGlass** [comment]
I don't like the removal of the `event-%classinfo%` also could be used by external addons using the testing suite.

I don't mind having test-%classinfo% but I prefer event-%classinfo% more. I have no say, so I don't want to review this.

**Comment by NotSoDelayed** [comment]
I agree, removing `event-*` without a grace period is a breaking change.

**Review by cheeezburga — APPROVED** [review]
Looks good!

**Review by sovdeeth — CHANGES_REQUESTED** [review]
seems good but no reason to remove the event values

File: src/main/java/ch/njol/skript/test/runner/EvtTestCase.java [hunk 1]
```diff
@@ -1,61 +1,45 @@
...
 	static {
 		if (TestMode.ENABLED) {
 			Skript.registerEvent("Test Case", EvtTestCase.class, SkriptTestEvent.class, "test %string% [when <.+>]")
-					.description("Contents represent one test case.")
-					.examples("")
-					.since("2.5");
-			EventValues.registerEventValue(SkriptTestEvent.class, Block.class, new Getter<Block, SkriptTestEvent>() {
+				.description("Contents represent one test case.")
+				.examples("")
+				.since("2.5");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/EvtTestCase.java:23]: 2 tabs

File: src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		"\tspawn zombie at test location",
+		"\tassert last spawned zombie's world is test world with \"zombie did not spawn in test world\""
+})
+@NoDoc
+public class ExprTestBlock extends SimpleExpression<Block> {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprTestBlock.class, Block.class, ExpressionType.SIMPLE,
+				"[the] test(-| )block");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java:30]: 2 tabs

File: src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	@Override
+	public Class<? extends Block> getReturnType() {
+		return Block.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "the test block";
+	}
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java:58]:
```suggestion
	}

```

File: src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"test \"example\":",
+		"\tset test block to air"
+})
+@NoDoc
+public class ExprTestLocation extends SimpleExpression<Location> {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprTestLocation.class, Location.class, ExpressionType.SIMPLE,
+				"[the] test(-| )location");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java:29]: 2t

File: src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	@Override
+	public Class<? extends Location> getReturnType() {
+		return Location.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "the test location";
+	}
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java:57]:
```suggestion
	}

```

File: src/main/java/ch/njol/skript/test/runner/ExprTestWorld.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		"\tspawn zombie at test location",
+		"\tassert last spawned zombie's world is test world with \"zombie did not spawn in test world\""
+})
+@NoDoc
+public class ExprTestWorld extends SimpleExpression<World> {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprTestWorld.class, World.class, ExpressionType.SIMPLE,
+				"[the] test(-| )world");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestWorld.java:30]: 2t

File: src/main/java/ch/njol/skript/test/runner/ExprTestWorld.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+	@Override
+	public Class<? extends World> getReturnType() {
+		return World.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return "the test world";
+	}
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestWorld.java:58]:
```suggestion
	}

```

File: src/main/java/ch/njol/skript/test/runner/ExprTestWorld.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+})
+@NoDoc
+public class ExprTestWorld extends SimpleExpression<World> {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprTestWorld.class, World.class, ExpressionType.SIMPLE,
+				"[the] test(-| )world");
+	}
+
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestWorld.java:32]:
```suggestion
```

File: src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+})
+@NoDoc
+public class ExprTestLocation extends SimpleExpression<Location> {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprTestLocation.class, Location.class, ExpressionType.SIMPLE,
+				"[the] test(-| )location");
+	}
+
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java:31]:
```suggestion
```

File: src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+})
+@NoDoc
+public class ExprTestBlock extends SimpleExpression<Block> {
+
+	static {
+		if (TestMode.ENABLED)
+			Skript.registerExpression(ExprTestBlock.class, Block.class, ExpressionType.SIMPLE,
+				"[the] test(-| )block");
+	}
+
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java:32]:
```suggestion
```

**Review by sovdeeth — CHANGES_REQUESTED** [review]
just some formatting

**Review by sovdeeth — APPROVED** [review]

File: src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Test Block")
+@Description("The block the testing is taking place at.")
+@Examples({
+	"test \"example\":",
+		"\tspawn zombie at test location",
+		"\tassert last spawned zombie's world is test world with \"zombie did not spawn in test world\""
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/test/runner/ExprTestBlock.java:22]: call me crazy, but "test block" is not used here at all?

File: src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Test Location")
+@Description("The location the testing is taking place at.")
+@Examples({
+	"test \"example\":",
+		"\tset test block to air"
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/test/runner/ExprTestLocation.java:21]: Oh... here it is?!?! I think you swapped test block/location examples.

**Review by Moderocky — APPROVED** [review]

**Pull request merged**